doc/src/bond_over_grpc.md

```diff
@@ -90,21 +96,141 @@ On the client side, the proxy stub establishes a connection to the server like t
+        }
+    }
+
+This service implementation is hooked up to a gRPC server as follows:
+
+    auto ioManager = std::make_shared<bond::ext::gRPC::io_manager>();
+    auto threadPool = std::make_shared<bond::ext::gRPC::thread_pool>();
+    const std::string server_address(Host + ":" + Port);
+
+    ExampleServiceImpl service;
+    bond::ext::gRPC::server_builder builder;
+    builder.SetThreadPool(threadPool);
+    const std::string server_address(Host + ":" + Port);
+    builder.AddListeningPort(server_address, grpc::InsecureServerCredentials());
+    builder.RegisterService(&service);
+    std::unique_ptr<bond::ext::gRPC::server> server(builder.BuildAndStart());
+
+At this point the server is ready to receive requests and route them to the
+service implemetation.
+
+On the client side, the proxy stub establishes a connection to the server like this:
+
+    auto ioManager = std::make_shared<bond::ext::gRPC::io_manager>();
+    auto threadPool = std::make_shared<bond::ext::gRPC::thread_pool>();
+    const std::string server_address(Host + ":" + Port);
+
+    Example::Client client(
+        grpc::CreateChannel(server_address, grpc::InsecureChannelCredentials()),
+        ioManager,
+        threadPool);
+
+The proxy stub can then be used to make calls to the server as follows:
+
+    ExampleRequest request;
+    // Fill in request fields here
+
+    bond::ext::gRPC::wait_callback<ExampleResponse> cb;
+    client.AsyncExampleMethod(&context, request, callback);
+
+    callback.wait();
+    ExampleResponse response = callback.response().Deserialize();
+    // Examine response here
+
+Note these APIs are significantly different from the APIs presented in the
+gRPC documentation; Bond-over-gRPC is attempting to provide a more
+straightforward API for aynchronous communication than gRPC currently presents
+in C++. The use of `wait_callback<T>` here is for illustrative purposes; any
+callback implementation with the same signature can be used. Note also the use
+of `bonded<T>` to wrap the request and response objects in several places;
+this allows for better control over the time of deserialization and also helps
+prevent slicing when using polymorphic Bond types. Finally, note that
+Bond-over-gRPC does not provide synchronous APIs in C++ by design.
+
+For more information about gRPC in C++, take a look at the
+[gRPC C++ tutorial](http://www.grpc.io/docs/tutorials/basic/c.html);
+however, keep in mind that the Bond-over-gRPC APIs diverge significantlty
+from those documented there.
+
+See also the following example:
+
+- `examples/cpp/grpc/helloworld`
+
+
+[^1] Technically, `Service` is a type alias for
+`ServiceCore<bond::ext::gRPC::thread_pool>`. The `ServiceCore<T>` template
+can be used to customize the service implementation to use a different
+thread pool implementation.
+
+[^2] Likewise, `Client` is a type alias for
+`ClientCore<bond::ext::gRPC::thread_pool>`. The `ClientCore<>` template
```

Comment:
Minor: there an inconsistency between `ServiceCore<T>` above and just `ClientCore<>` here.